platform/pc/service.c

```diff
@@ -501,6 +542,24 @@ void init_service(u64 rdi, u64 rsi)
     switch_stack(stack_location, init_service_new_stack);
 }
 
+void pvh_start(hvm_start_info start_info)
+{
```

Comment:
I tried running this branch on my Mac with KERNEL_BOOT=1, but it crashes here because the function preamble contains the stack smashing guard which references a virtual address that hasn't been mapped yet (e.g. 0xffffffff80131030) and so the unhandled page fault just causes qemu to exit with no messages. I briefly looked for an annotation that can turn it off by function but haven't found one yet. Also not sure if this is just a clang specific problem.